### PR TITLE
Compress comments in make_image.

### PR DESCRIPTION
**Pull request by anntzer**
make_image is a huge function; shorten it (in physical screen space) a
bit by rewriting/rewrapping some comments, and a few small stylistic
changes.

## PR Checklist

- [ ] Has pytest style unit tests (and `pytest` passes).
- [ ] Is [Flake 8](https://flake8.pycqa.org/en/latest/) compliant (run `flake8` on changed files to check).
- [ ] New features are documented, with examples if plot related.
- [ ] Documentation is sphinx and numpydoc compliant (the docs should [build](https://matplotlib.org/devel/documenting_mpl.html#building-the-docs) without error).
- [ ] Conforms to Matplotlib style conventions (install `flake8-docstrings` and run `flake8 --docstring-convention=all`).
- [ ] New features have an entry in `doc/users/next_whats_new/` (follow instructions in README.rst there).
- [ ] API changes documented in `doc/api/next_api_changes/` (follow instructions in README.rst there).

### CODE REVIEW

File: lib/matplotlib/image.py [hunk 1]
```diff
@@ -456,30 +447,25 @@ def _make_image(self, A, in_bbox, out_bbox, clip_bbox, magnification=1.0,
...
 
-                # used to rescale the raw data to [offset, 1-offset]
-                # so that the resampling code will run cleanly.  Using
-                # dyadic numbers here could reduce the error, but
-                # would not full eliminate it and breaks a number of
-                # tests (due to the slightly different error bouncing
-                # some pixels across a boundary in the (very
+                # Rescale the raw data to [offset, 1-offset] so that the
+                # resampling code will run cleanly.  Using dyadic numbers here
+                # could reduce the error, but would not full eliminate it and
```
Comment (QuLogic) [inline lib/matplotlib/image.py:452]:
```suggestion
                # could reduce the error, but would not fully eliminate it and
```

File: lib/matplotlib/image.py [hunk 2]
```diff
@@ -568,17 +547,15 @@ def _make_image(self, A, in_bbox, out_bbox, clip_bbox, magnification=1.0,
             if A.ndim == 2:
                 alpha = self._get_scalar_alpha()
                 alpha_channel = output[:, :, 3]
-                alpha_channel[:] = np.asarray(
-                    np.asarray(alpha_channel, np.float32) * out_alpha * alpha,
-                    np.uint8)
+                alpha_channel[:] = (
+                    alpha_channel.astype(np.float32) * out_alpha * alpha)
```
Comment (QuLogic) [inline lib/matplotlib/image.py:551]: This doesn't need the cast to uint8?

File: lib/matplotlib/image.py (hunk 1 above)
Comment (anntzer) [reply lib/matplotlib/image.py]: indeed

File: lib/matplotlib/image.py (hunk 2 above)
Comment (anntzer) [reply lib/matplotlib/image.py]: no, added a comment to clarify.

**Review by jklymak — APPROVED** [review]

**Pull request merged**